### PR TITLE
Add archive file to cache files that passed checksum validation (#600)

### PR DESCRIPTION
**Pull request by benbou8231**
This is my attemp to implement #600.

Instead of using `--checksum`, the user can now use `--checksum-archive`. 
This will create a file named `_checksum_archive.txt` and write a line to it every time a checksum validation succeeds. Subsequent runs will skip the files listed in `_checksum_archive.txt`.

To implement this feature and ensure that the tests related to my changes pass, I had to increase the `max-args` to 24 in `pyproject.toml`.

In conclusion, this solution enhances the efficiency of checksum validations by skipping previously validated files.
I look forward to your feedback and suggestions for any improvements.

### CODE REVIEW

**Comment by benbou8231** [comment]
I just resolved the merge conflict introduced by #650
